pkg/collect/cluster_resources.go

```diff
@@ -431,6 +441,48 @@ func crds(ctx context.Context, client *apiextensionsv1beta1clientset.Apiextensio
 	return b, nil
 }
 
+func crs(ctx context.Context, client *apiextensionsv1beta1clientset.ApiextensionsV1beta1Client) (map[string][]byte, map[string]string) {
+	customResources := make(map[string][]byte)
+	errorList := make(map[string]string)
+	customResourceItems := struct {
+		metav1.TypeMeta `json:",inline"`
+		metav1.ListMeta `json:"metadata,omitempty"`
+		Items           []map[string]interface{} `json:"items"`
+	}{}
+	crds, err := client.CustomResourceDefinitions().List(ctx, metav1.ListOptions{})
+	if err != nil {
+		errorList[crds.Kind] = err.Error()
+		return customResources, errorList
+	}
+	// Loop through CRDs to fetch the CRs
+	for _, v := range crds.Items {
+		data := client.RESTClient().Get().AbsPath("/apis/" + v.Spec.Group + "/" + v.Spec.Version).Do(ctx)
+		apiResourceListObj, err := data.Get()
+		gv := v.Spec.Group + "-" + v.Spec.Version
+		if err != nil {
+			errorList[gv] = err.Error()
+		}
+		apiResourceList, _ := apiResourceListObj.(*metav1.APIResourceList)
+		groupVersion := apiResourceList.GroupVersion
+		for _, v := range apiResourceList.APIResources {
+			customResourceName := v.Name
+			if customResourceName != "" && !strings.ContainsAny(customResourceName, "/") {
+				gv = gv + "-" + customResourceName + ".json"
```

Comment:
Doesn't this grow `gv` on each iteration through the loop? Initially it is `group-version`. The first iteration makes it `group-version-crname1.json`. The next iteration makes it `group-version-crname1.json-crname2.json`...

How about a new variable name for the "filename"?

Reply:
Thanks, Jon. Noted!